ReactiveCocoaTests/Swift/SignalSpec.swift

```diff
@@ -218,48 +153,57 @@ class SignalSpec: QuickSpec {
 			it("should deallocate after erroring") {
 				let testScheduler = TestScheduler()
 				weak var weakSignal: Signal<(), TestError>?
+				var error: Bool = false
 				
 				// Use an inner closure to help ARC deallocate things as we
 				// expect.
 				let test: () -> () = {
 					let (signal, observer) = Signal<(), TestError>.pipe()
 					weakSignal = signal
+
 					testScheduler.schedule {
 						observer.sendFailed(TestError.Default)
 					}
+
+					signal.observeFailed { _ in error = true }
 				}
 				test()
-				
-				expect(weakSignal).toNot(beNil())
-				
-				testScheduler.run()
+
+				expect(error).to(beFalsy())
 				expect(weakSignal).to(beNil())
+				testScheduler.run()
+				expect(error).to(beTruthy())
 			}
 
 			it("should deallocate after completing") {
 				let testScheduler = TestScheduler()
 				weak var weakSignal: Signal<(), TestError>?
+				var completed: Bool = false
 				
 				// Use an inner closure to help ARC deallocate things as we
 				// expect.
 				let test: () -> () = {
 					let (signal, observer) = Signal<(), TestError>.pipe()
 					weakSignal = signal
+
 					testScheduler.schedule {
 						observer.sendCompleted()
 					}
+
+					signal.observeCompleted { completed = true }
 				}
 				test()
 				
-				expect(weakSignal).toNot(beNil())
-
-				testScheduler.run()
+				expect(completed).to(beFalsy())
```

Comment:
This is redundant anyway, it should be covered by another test. What this test should cover is the deallocation, which it no longer does?